### ISSUE TITLE
Operators in indicator call signatures

### ISSUE THREAD

**Issue by Zeitsperre** [Ouranosinc/xclim#389]
Follow-up on a discussion point raised in https://github.com/Ouranosinc/xclim/pull/379#discussion_r386490720

When creating new indices based on the existing set, there are instances where it would be useful to be able to set a threshold with a different operator. What this means is that for indices that use `> | <` in some definitions and `>= | <=` in others, it would make for a more versatile system if when wrapping an existing base indicator, we were able to specify the type of operation we can use to trigger the conditions.

In other words, the proposal is to add the following kwargs to all indices based on conditional thresholds:
```
f = indices.tropical_nights(tas, thresh="20 degC", operator="gteq | gt | > | >=")
```
This would result in wrapped indices that look more like this:
```
R75p = partial(indices.days_over_precip_thresh, thresh="1 mm/day", operator=">")
```

**Comment by aulemahal** [comment]
I agree. For code clarity, we should have a function in `generic.py`:
```python
def threshold(left, op, right):
    from xarray.core.ops import get_op
    if op in binary_ops:
        op = binary_ops[op]
    elif op in binary_ops.values():
        pass
    else:
        raise ValueError(f"Operation `{op}` not recognized.")

    func = getattr(left, "_binary_op")(get_op(op))
    return func(left, right)
```
(code copied from `xclim.indices.generic.threshold_count`, which I thought was useless up to now)
So we can use it like:
```
threshold(tas, op, thresh_tas)
```
